### ISSUE TITLE
Native support for data type of `BigInt` 

### ISSUE THREAD

**Issue by dotansimha** [graphprotocol/graph-client#90]

**Comment by matthewlilley** [comment]
Yeah, I'm a bit hessitent to enforce BigInt just yet. There's good but not enough support for all users, yet. We use JSBI to represent BigInt in our dApps, which I think would be a great middle ground.

**Comment by ardatan** [comment]
Available in the latest release!

**Comment by matthewlilley** [comment]
@ardatan Is there some way to disable this? We don't use native bigint for a few reasons, one being that we aren't able to serialize it, and do a lot of SSR. We use JSBI in our SDKs also to provide maximum browser coverage. This is a bit of a pain for us atm and preventing updating.

**Comment by matthewlilley** [comment]
Worked around for now, but would be nice if we were able to transform BigInt & BigDecimal ourselves, depending on needs.

**Comment by ardatan** [comment]
@matthewlilley Graph Client uses `json-bigint-patch` to handle BigInt parsing and serialization. BigInts are first class citizens of JavaScript today and supported by the major browsers. 
Could you explain more about the issue you have? Maybe we can help you.

**Comment by matthewlilley** [comment]
@ardatan I'm unable to build with type safety enabled now.

![Screenshot_804](https://user-images.githubusercontent.com/6063350/189794893-9e5b53b6-f1c6-488b-8751-e4281f7d2201.png)

If I cast to BigInt, like so

![Screenshot_805](https://user-images.githubusercontent.com/6063350/189795355-688b9964-0483-4d18-9654-5ca45e99aa16.png)

Error: Do not know how to serialize a BigInt

**Comment by matthewlilley** [comment]
I think it's a bad to assume and even work enforce that everyone would like native bigint support.

And in a case like this it makes even less sense, because it's a timestamp which will never be BigInt, it's a unsigned 32-bit integer...

**Comment by matthewlilley** [comment]
Maybe type BigIntIsh = bigint | string?

**Comment by matthewlilley** [comment]
This makes too many assumptions. It's parsing ERC20 decimals to BigInt too, this is a unsigned 8 bit integer, it makes no sense to ever be represented as a BigInt but we're forced with it.

**Comment by azf20** [comment]
Agree with @matthewlilley, often users will use BigInt simply for convenience / as a default, and users don't expect the native type

**Comment by matthewlilley** [comment]
> Agree with @matthewlilley, often users will use BigInt simply for convenience / as a default, and users don't expect the native type

To contiue this conversation, we plan to eventually migrate to native bigint, but given the speed at which we can iterate subgraphs this would need to happen over a long time period, so support for incremental adoption of native bigint , or other transforms is critical.
